### PR TITLE
Stop serving Observatory by default

### PR DESCRIPTION
**Pull request by bkonyi**
Observatory can still be enabled by providing `--serve-observatory` or invoking the `_serveObservatory` private service RPC via web socket or HTTP.

Related to https://github.com/dart-lang/sdk/issues/50233

### CODE REVIEW

**Comment by zanderso** [comment]
The test failures look related.

File: packages/flutter_tools/test/integration.shard/flutter_attach_test.dart [hunk 1]
```diff
@@ -197,24 +197,20 @@ void main() {
...
+          serveObservatory: true,
         );
         expect(await isObservatoryAvailable(), true);
     });
 
     testWithoutContext('enables Observatory on attach', () async {
-      await flutterRun.run(withDebugger: true, serveObservatory: false);
+      await flutterRun.run(withDebugger: true);
       // Bail out if Observatory is still served by default in the VM.
       if (await isObservatoryAvailable()) {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/flutter_attach_test.dart:208]: is observatory still served by default by the vm?

Comment (bkonyi) [reply packages/flutter_tools/test/integration.shard/flutter_attach_test.dart:208]: No, it's not.

File: packages/flutter_tools/test/integration.shard/flutter_attach_test.dart [hunk 2]
```diff
@@ -197,24 +197,20 @@ void main() {
...
         return;
       }
```
Comment (christopherfujino) [inline packages/flutter_tools/test/integration.shard/flutter_attach_test.dart:210]: Can we update this then to:
```suggestion
      expect(await isObservatoryAvailable(), isFalse);
```

Comment (bkonyi) [reply packages/flutter_tools/test/integration.shard/flutter_attach_test.dart:210]: No, since the change in the VM hasn't rolled in yet.

Comment (christopherfujino) [reply packages/flutter_tools/test/integration.shard/flutter_attach_test.dart:210]: Which change?

Comment (bkonyi) [reply packages/flutter_tools/test/integration.shard/flutter_attach_test.dart:210]: It was reverted in https://github.com/dart-lang/sdk/commit/5a8ddc0756684e8ebfbc0edfbfc6046755b8394a, which will reland once this change lands.

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/integration.shard/flutter_attach_test.dart [hunk 3]
```diff
@@ -197,24 +197,22 @@ void main() {
...
           withDebugger: true,
-          // TODO(bkonyi): uncomment once Observatory is disabled by default
-          // See https://github.com/dart-lang/sdk/issues/50233
-          // serveObservatory: true,
+          serveObservatory: true,
         );
         expect(await isObservatoryAvailable(), true);
     });
 
     testWithoutContext('enables Observatory on attach', () async {
```
Comment (zanderso) [inline packages/flutter_tools/test/integration.shard/flutter_attach_test.dart:205]: Should this test have a new name?

Comment (bkonyi) [reply packages/flutter_tools/test/integration.shard/flutter_attach_test.dart:205]: No, this test was named this way in anticipation of Observatory being disabled by default.

**Pull request merged**